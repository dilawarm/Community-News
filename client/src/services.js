// @flow
import axios from 'axios';

export class Nyhetssak {
  saksId: number;
  overskrift: string;
  innhold: string;
  tidspunkt: string;
  bilde: string;
  kategori: string;
  viktighet: boolean;
  rating: number;
  brukerId: number;

  constructor(saksId: number, overskrift: string, innhold: string, tidspunkt: string, bilde: string, kategori: string, viktighet: boolean, rating: number, brukerId: number) {
      this.saksId = saksId;
      this.overskrift = overskrift;
      this.innhold = innhold;
      this.tidspunkt = tidspunkt;
      this.bilde = bilde;
      this.kategori = kategori;
      this.viktighet = viktighet;
      this.rating = rating;
      this.brukerId = brukerId;
  }
}

class NyhetssakService {
    getSaker() {
        return axios.get<Nyhetssak[]>('http://localhost:8080/nyhetssaker').then(response => response.data);
    }

    getSakKat(kategori: string) {
        return axios.get<Nyhetssak[]>('http://localhost:8080/nyhetssaker/' + kategori).then(response => response.data);
    }

    getSakKatId(kategori: string, id: number) {
        return axios.get<Nyhetssak[]>('http://localhost:8080/nyhetssaker/' + kategori + "/" + id).then(response => response.data[0]);
    }

    postSak(overskrift: string, innhold: string, bilde: string, kategori: string, viktighet: boolean, brukerId: number) {
        return axios.post<{}, Nyhetssak>('http://localhost:8080/nyhetssaker', {
            "overskrift": overskrift,
            "innhold": innhold,
            "bilde": bilde,
            "kategori": kategori,
            "viktighet": viktighet,
            "brukerId": brukerId
        }).then(response => response.data);
    }

    deleteSak(id: number) {
        return axios.delete<Nyhetssak, void>('http://localhost:8080/nyhetssaker/' + id).then(response => response.data);
    }

    upvote(id: number) {
        return axios.put<Nyhetssak, void>('http://localhost:8080/nyhetssaker/' + id + "/upvote").then(response => response.data);
    }

    downvote(id: number) {
        return axios.put<Nyhetssak, void>('http://localhost:8080/nyhetssaker/' + id + "/downvote").then(response => response.data);
    }

    livefeed() {
        return axios.get<Nyhetssak[]>('http://localhost:8080/livefeed').then(response => response.data);
    }

    updateSak(id: number, overskrift: string, innhold: string, bilde: string, kategori: string, viktighet: boolean) {
        return axios.put<{}, Nyhetssak>('http://localhost:8080/nyhetssaker/rediger/' + id, {
            "overskrift": overskrift,
            "innhold": innhold,
            "bilde": bilde,
            "kategori": kategori,
            "viktighet": viktighet,

        }).then(response => response.data);
    }

    getSakBruker(id: number) {
        return axios.get<Nyhetssak[]>('http://localhost:8080/nyhetssaker/kategorier/MineSaker/' + id).then(response => response.data);
    }

    getForfatter(id: number) {
        return axios.get<{brukernavn: string}>('http://localhost:8080/nyhetssaker/kategori/brukernavn/' + id).then(response => response.data);
    }

    getSokSak(soketekst: string) {
        return axios.get<Nyhetssak[]>('http://localhost:8080/nyhetssaker/kategori/sokeSak/' + soketekst).then(response => response.data);   
    }
}

export let nyhetssakService = new NyhetssakService();

export class Kommentar {
  kommId: number;
  nick: string;
  kommentar: string;
  saksId: number;
  tidspunkt: string;
}

class KommentarService {
    getKommentarer(kategori: string, id: number) {
        return axios.get<Kommentar[]>('http://localhost:8080/nyhetssaker/' + kategori + "/" + id + "/kommentarer").then(response => response.data);
    }

    postKommentarer(id: number, kommentar: string, nick: string) {
        return axios.post<{}, Kommentar>('http://localhost:8080/nyhetssaker/' + id, {
            "kommentar": kommentar,
            "nick": nick
        }).then(response => response.data);
    }

    deleteKommentar(id: number) {
        return axios.delete<Kommentar, void>('http://localhost:8080/nyhetssaker/kommentarer/' + id).then(response => response.data);
    }
}

export let kommentarService = new KommentarService();

export class Bruker {
    brukerId: number;
    brukernavn: string;

    constructor(brukerId: number, brukernavn: string) {
        this.brukerId = brukerId;
        this.brukernavn = brukernavn;
    }
}

class BrukerService {
    getOne(brukernavn: string) {
        return axios.get<Bruker[]>('http://localhost:8080/brukere/' + brukernavn).then(response => response.data[0]);
    }

    login(brukernavn: string, passord: string) {
        return axios.post<{}, {jwt: string}>('http://localhost:8080/login', {
            "brukernavn": brukernavn,
            "passord": passord
        }).then(response => response.data);
    }

    registrer(brukernavn: string, passord: string) {
        return axios.post<{}, Bruker>('http://localhost:8080/registrer', {
            "brukernavn": brukernavn,
            "passord": passord
        }).then(response => response.data)
    }

    postToken(brukernavn: string, token: string) {
        return fetch("http://localhost:8080/token",
  	    {
        method: "POST",
        headers: {
    	    "Content-Type": "application/json; charset=utf-8",
    	    "x-access-token": token
	    },
        body: JSON.stringify({"brukernavn": brukernavn})
        })
        .then(response => response.json())
    }
}

export let brukerService = new BrukerService();
