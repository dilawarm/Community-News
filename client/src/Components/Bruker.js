// @flow

import * as React from 'react';
import Button from 'react-bootstrap/Button';
import Form from 'react-bootstrap/Form';
import { Component } from 'react-simplified';
import {Alert} from '../widgets';
import {Bruker, brukerService} from '../services';
import {Navigation} from './staticComponents';
import { createHashHistory } from 'history';

const history = createHashHistory();

export class Login extends Component {
  brukernavn: string = '';
  passord: string = '';
  inn_bruker: Bruker | any = null;

  render() {
    return <>
    <div className="mx-auto w-75">
    <Form>
  <Form.Group controlId="exampleForm.ControlInput1">
    <Form.Label>Brukernavn</Form.Label>
    <Form.Control 
    type="text" 
    value={this.brukernavn}
    onChange={(event: SyntheticInputEvent<HTMLInputElement>) => (this.brukernavn = event.target.value)}
    />
  </Form.Group>
  <Form.Group controlId="exampleForm.ControlInput1">
    <Form.Label>Passord</Form.Label>
    <Form.Control
    id ="password"
    type="password" 
    value={this.passord}
    onChange={(event: SyntheticInputEvent<HTMLInputElement>) => (this.passord = event.target.value)}
    />
  </Form.Group>
  <Button variant="primary" onClick={this.logg_in}>Logg inn</Button>
</Form>
</div>
    </>
  }

  logg_in() {
    if (this.brukernavn == "") {
      Alert.danger("Ugyldig brukernavn");
      return;
    }
    if (this.passord == "") {
      Alert.danger("Ugyldig passord");
      return;
    }

    let token: string;
    brukerService
      .login(this.brukernavn, this.passord)
      .then(json => {
        token = json.jwt;
        localStorage.setItem("token", json.jwt);
    	  //console.log(JSON.stringify(json));
        brukerService
          .getOne(this.brukernavn)
          .then(json => {
            let nav: any = Navigation.instance();
            nav.mounted(new Bruker(json.brukerId, json.brukernavn))
          })
          .catch((error: Error) => Alert.danger(error.message));
      })
      .then(() => {
        brukerService
          .postToken(this.brukernavn, token)
          .then(json => localStorage.setItem("token", json.jwt))
          .catch((error: Error) => Alert.danger(error.message));
     })
     .then(history.push("/"))
     .catch((error: Error) => Alert.danger(error.message));
  }
}

export class Registrer extends Component {
  brukernavn = '';
  passord = '';

  render() {
    return <>
    <div className="mx-auto w-75">
    <Form>
  <Form.Group controlId="exampleForm.ControlInput1">
    <Form.Label>Brukernavn</Form.Label>
    <Form.Control
    type="text" 
    value={this.brukernavn}
    onChange={(event: SyntheticInputEvent<HTMLInputElement>) => (this.brukernavn = event.target.value)}
    />
  </Form.Group>
  <Form.Group controlId="exampleForm.ControlInput1">
    <Form.Label>Passord</Form.Label>
    <Form.Control
    type="password" 
    value={this.passord}
    onChange={(event: SyntheticInputEvent<HTMLInputElement>) => (this.passord = event.target.value)}
    />
  </Form.Group>
  <Button variant="success" onClick={this.register}>Registrer bruker</Button>
</Form>
</div>
    </>
  }

  register() {
    if (this.brukernavn == "") {
      Alert.danger("Ugyldig brukernavn");
      return;
    }
    if (this.passord == "") {
      Alert.danger("Ugyldig passord");
      return;
    }

    brukerService
      .registrer(this.brukernavn, this.passord)
      .then(history.push("/"))
      .then(Alert.success("Brukeren er registrert! Du kan logge inn nÃ¥."))
      .catch((error: Error) => console.log("")); 
  }
}