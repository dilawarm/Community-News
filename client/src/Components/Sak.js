// @flow

import * as React from 'react';
import Nav from 'react-bootstrap/Nav';
import Card from 'react-bootstrap/Card';
import Button from 'react-bootstrap/Button';
import Form from 'react-bootstrap/Form';
import Col from 'react-bootstrap/Col';
import { Component } from 'react-simplified';
import {Alert} from '../widgets';
import {nyhetssakService, kommentarService, Kommentar, Nyhetssak, Bruker} from '../services';
import {Navigation} from './staticComponents';
import { createHashHistory } from 'history';

const history = createHashHistory();

export class Sak extends Component<{ match: { params: { id: number, kategori: string } } }> {
  nick: string = '';
  kommentar: string = '';
  sak: Nyhetssak | any = null;
  kommentarer: Kommentar[] = [];
  inn_bruker: Bruker | any = null;
  brukerId: number = 0;
  forfatter: string = '';

  render() {
    if (this.sak) {
      if (this.inn_bruker) {
        if (this.inn_bruker.brukerId == this.brukerId) {
          return <>
    <Form.Row>
    <Nav.Link>
    <Button id="slett" variant="danger" onClick={this.delete}>Slett nyhetsartikkel</Button>
    </Nav.Link>
    <Nav.Link href={"#/rediger/"+this.props.match.params.kategori+"/"+this.props.match.params.id}>
    <Button variant="success">Rediger nyhetsartikkel</Button>
    </Nav.Link>
    </Form.Row>
    <Card className="bg-light text-black" className="mx-auto w-75">
        <Card.Img id="img_article" src={this.sak.bilde}/>
           <Card.Title><h1>{this.sak.overskrift}</h1></Card.Title>
           <Card.Text id="forfatter">Forfatter: {this.forfatter}</Card.Text>
           <Card.Text id="forfatter">Kategori: {this.props.match.params.kategori}</Card.Text>
           <Card.Text id="innhold">{this.sak.innhold}</Card.Text>
      <Card.Text id="tid">Sist oppdatert: {this.sak.tidspunkt.substring(0, 10) + " " + this.sak.tidspunkt.substring(11, 16)}</Card.Text>
      <Card.Text>Rating: <p id="rating"><b>{this.sak.rating}</b></p></Card.Text>
      <Form.Row>
      <Button id="upvote" variant="primary" onClick={this.upvote} style={{width: "100px"}}>Upvote</Button>
      <Button id="downvote" variant="danger" onClick={this.downvote} style={{width: "100px"}}>Downvote</Button>
      </Form.Row>
</Card>
    <Form className="mx-auto w-75">
    <Card.Title><h2>Kommentarer</h2></Card.Title>
  <Form.Row>
    <Col>
      <Form.Control
      placeholder="Kommentar"
      type="text"
      value = {this.kommentar}
      onChange = {(event: SyntheticInputEvent<HTMLInputElement>) => (this.kommentar = event.target.value)} 
      />
    </Col>
  </Form.Row>
  <Button id="comment-button" variant="primary" onClick={this.add_b}>Kommenter</Button>
</Form>
  {this.kommentarer.map(kommentar => (
    <Card id="comment" className="bd-callout bd-callout-info mx-auto w-75">
           <Card.Title id="nick">{kommentar.nick}</Card.Title>
           <Card.Text>{kommentar.kommentar}</Card.Text>
           <Button variant="danger" onClick={ () => this.del_comment(kommentar.kommId) } style={{width: "150px"}}>Slett kommentar</Button> 
</Card>
  ))}
    </> 
        }
        else {
          return <>
    <Card className="bg-light text-black" className="mx-auto w-75">
        <Card.Img id="img_article" src={this.sak.bilde}/>
           <Card.Title><h1>{this.sak.overskrift}</h1></Card.Title>
           <Card.Text id="forfatter">Forfatter: {this.forfatter}</Card.Text>
           <Card.Text id="forfatter">Kategori: {this.props.match.params.kategori}</Card.Text>
           <Card.Text id="innhold">{this.sak.innhold}</Card.Text>
      <Card.Text id="tid">Sist oppdatert: {this.sak.tidspunkt.substring(0, 10) + " " + this.sak.tidspunkt.substring(11, 16)}</Card.Text>
      <Card.Text>Rating: <p id="rating"><b>{this.sak.rating}</b></p></Card.Text>
      <Form.Row>
      <Button id="upvote" variant="primary" onClick={this.upvote} style={{width: "100px"}}>Upvote</Button>
      <Button id="downvote" variant="danger" onClick={this.downvote} style={{width: "100px"}}>Downvote</Button>
      </Form.Row>
</Card>
    <Form className="mx-auto w-75">
    <Card.Title><h2>Kommentarer</h2></Card.Title>
  <Form.Row>
    <Col>
      <Form.Control
      placeholder="Kommentar"
      type="text"
      value = {this.kommentar}
      onChange = {(event: SyntheticInputEvent<HTMLInputElement>) => (this.kommentar = event.target.value)} 
      />
    </Col>
  </Form.Row>
  <Button id="comment-button" variant="primary" onClick={this.add_b}>Kommenter</Button>
</Form>

  {this.kommentarer.map(kommentar => (
    <Card id="comment" className="bd-callout bd-callout-info mx-auto w-75">
           <Card.Title id="nick">{kommentar.nick}</Card.Title>
           <Card.Text>{kommentar.kommentar}</Card.Text>
</Card>
  ))}
    </> 
        }
      }
      else {
        return <>
    <Card className="bg-light text-black" className="mx-auto w-75">
        <Card.Img id="img_article" src={this.sak.bilde}/>
           <Card.Title><h1>{this.sak.overskrift}</h1></Card.Title>
           <Card.Text id="forfatter">Forfatter: {this.forfatter}</Card.Text>
           <Card.Text id="forfatter">Kategori: {this.props.match.params.kategori}</Card.Text>
           <Card.Text id="innhold">{this.sak.innhold}</Card.Text>
      <Card.Text id="tid">Sist oppdatert: {this.sak.tidspunkt.substring(0, 10) + " " + this.sak.tidspunkt.substring(11, 16)}</Card.Text>
      <Card.Text>Rating: <p id="rating"><b>{this.sak.rating}</b></p></Card.Text>
      <Form.Row>
      <Button id="upvote" variant="primary" onClick={this.upvote} style={{width: "100px"}}>Upvote</Button>
      <Button id="downvote" variant="danger" onClick={this.downvote} style={{width: "100px"}}>Downvote</Button>
      </Form.Row>
</Card>
    <Form className="mx-auto w-75">
    <Card.Title><h2>Kommentarer</h2></Card.Title>
  <Form.Row>
  <Col>
      <Form.Control
      placeholder="Nick"
      type="text"
      value = {this.nick}
      onChange = {(event: SyntheticInputEvent<HTMLInputElement>) => (this.nick = event.target.value)} 
      />
    </Col>
    <Col>
      <Form.Control
      placeholder="Kommentar"
      type="text"
      value = {this.kommentar}
      onChange = {(event: SyntheticInputEvent<HTMLInputElement>) => (this.kommentar = event.target.value)} 
      />
    </Col>
  </Form.Row>
  <Button id="comment-button" variant="primary" onClick={this.add}>Kommenter</Button>
</Form>

  {this.kommentarer.map(kommentar => (
    <Card id="comment" className="bd-callout bd-callout-info mx-auto w-75">
           <Card.Title id="nick">{kommentar.nick}</Card.Title>
           <Card.Text>{kommentar.kommentar}</Card.Text>
</Card>
  ))}
    </> 
      }
    }
    else {
      return (
        <div></div>
      )
    }
  }

  mounted() {
    nyhetssakService
      .getSakKatId(this.props.match.params.kategori, this.props.match.params.id)
      .then(sak => {
        this.sak = sak;
        let d: any = document.getElementById('rating');
        if (this.sak.rating < 0) {
          d.style.color = "#8a0811";
        }
        else {
          d.style.color = "#035c1a";
        }
        this.brukerId = this.sak.brukerId;
      })
      .catch((error: Error) => console.log(""));
    
    nyhetssakService
      .getForfatter(this.props.match.params.id)
      .then(res => this.forfatter = res[0].brukernavn)
      .catch((error: Error) => console.log(""));
    
    
    kommentarService
      .getKommentarer(this.props.match.params.kategori, this.props.match.params.id)
      .then(kommentarer => (this.kommentarer = kommentarer))
      .catch((error: Error) => Alert.danger(error.message));
    
    let nav: any = Navigation.instance();
    this.inn_bruker = nav.inn_bruker;
  }

  add_b() {
    kommentarService
      .postKommentarer(this.props.match.params.id, this.kommentar, this.inn_bruker.brukernavn)
      .then(this.mounted())
      .catch((error: Error) => Alert.danger(error.message));
    
    this.kommentar = "";
    let s: any = Sak.instance()
    s.mounted();
  }

  add() {
    kommentarService
      .postKommentarer(this.props.match.params.id, this.kommentar, this.nick)
      .then(this.mounted())
      .catch((error: Error) => Alert.danger(error.message));
    
    this.nick = "";
    this.kommentar = "";
    let s: any = Sak.instance();
    s.mounted();
  }

  delete() {
    nyhetssakService
      .deleteSak(this.props.match.params.id)
      .then(() => {
        history.push("/");
        Alert.danger("Sak slettet!");
      })
      .catch((error: Error) => Alert.danger(error.message));
  }

  upvote() {
    nyhetssakService
      .upvote(this.props.match.params.id)
      .then(() => {
        let s: any = Sak.instance();
        s.mounted();
      })
      .catch((error: Error) => Alert.danger(error.message));

    let d: any = document.getElementById('upvote');
    d.disabled = 'disabled';
  }

  downvote() {
    nyhetssakService
      .downvote(this.props.match.params.id)
      .then(() => {
        let s: any = Sak.instance();
        s.mounted();
      })
      .catch((error: Error) => Alert.danger(error.message));
    
    let d: any = document.getElementById('downvote');
    d.disabled = 'disabled';
  }

  del_comment(kommId: number) {
    kommentarService 
      .deleteKommentar(kommId)
      .then(this.mounted())
      .catch((error: Error) => Alert.danger(error.message));
    
    let s: any = Sak.instance();
    s.mounted();
  }
}
