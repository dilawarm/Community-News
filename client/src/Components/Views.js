import Navbar from 'react-bootstrap/Navbar';
import Nav from 'react-bootstrap/Nav';
import Card from 'react-bootstrap/Card';
import Image from 'react-bootstrap/Image';
import Button from 'react-bootstrap/Button';
import Form from 'react-bootstrap/Form';
import Col from 'react-bootstrap/Col';
import CardGroup from 'react-bootstrap/CardGroup';
import ReactDOM from 'react-dom';
import * as React from 'react';
import { Component, sharedComponentData } from 'react-simplified';
import { HashRouter, Route, NavLink } from 'react-router-dom';
import {Alert} from '../widgets';
import {Nyhetssak, nyhetssakService, Kommentar, kommentarService, Bruker, brukerService} from '../services';
import {Navigation} from './staticComponents';
import socketIOClient from 'socket.io-client';

export class Forside extends Component {
  saker: Nyhetssak[] = [];

  render() {
    //let nysaker = saker.filter(sak => sak.viktighet == true);
    if (this.saker) {
      return <div class="grid-container">
      {this.saker.map(sak => (
        <div className="mx-auto w-75">
          <a href={"#/kategori"+"/"+sak.kategori+"/"+sak.saksId}>
            <Card id="article">
            <Card.Img src={sak.bilde} width="945" height="400"/>
            <Card.Body>
              <Card.Title><h2 id="overskrift">{sak.overskrift}</h2></Card.Title>
              <Card.Text><p id="tid">Sist oppdatert: {sak.tidspunkt.substring(0,10) + " " + sak.tidspunkt.substring(11, 16)}</p></Card.Text>
            </Card.Body>
          </Card>
          </a>
        </div>
      ))}
    </div>
    }
    else {
      return (
        <div>Laster forside...</div>
      )
    }
  }

  
  mounted() {
    nyhetssakService
      .getSaker()
      .then(saker => (this.saker = saker))
      .catch((error: Error) => Alert.danger(error.message));
  }
}

export class Sakliste extends Component<{ match: { params: { kategori: string } } }> {
  saker: Nyhetssak[] = [];
  render() {
    if (this.saker) {
      return <div class="grid-container">
      {this.saker.map(sak => (
        <div className="mx-auto w-75">
        <a href={"#/kategori"+"/"+sak.kategori+"/"+sak.saksId}>
  <Card id="article">
  <Card.Img src={sak.bilde} width="945" height="400"/>
  <Card.Body>
    <Card.Title><h2 id="overskrift">{sak.overskrift}</h2></Card.Title>
    <Card.Text><p id="tid">Sist oppdatert: {sak.tidspunkt.substring(0,10) + " " + sak.tidspunkt.substring(11, 16)}</p></Card.Text>
  </Card.Body>
</Card>
</a>
</div>
      ))}
    </div>
    }
    else {
      return (
        <div> Laster kategori {this.props.match.params.kategori}</div>
      )
    }
  }

  mounted() {
    nyhetssakService
      .getSakKat(this.props.match.params.kategori)
      .then(saker => (this.saker = saker))
      .catch((error: Error) => Alert.danger(error.message));
  }
}

export class MineSaker extends Component<{ match: { params: { id: number } } }> {
  saker: Nyhetssak[] = [];

  render() {
    console.log(this.saker);
    if (this.saker) {
      return <div class="grid-container">
      {this.saker.map(sak => (
        <div className="mx-auto w-75">
        <a href={"#/kategori"+"/"+sak.kategori+"/"+sak.saksId}>
  <Card id="article">
  <Card.Img src={sak.bilde} width="945" height="400"/>
  <Card.Body>
    <Card.Title><h2 id="overskrift">{sak.overskrift}</h2></Card.Title>
    <Card.Text><p id="tid">Sist oppdatert: {sak.tidspunkt.substring(0,10) + " " + sak.tidspunkt.substring(11, 16)}</p></Card.Text>
  </Card.Body>
</Card>
</a>
</div>
      ))}
      </div>
    }
    else {
      return (
        <div> Laster dine saker... </div>
      )
    }
  }

  mounted() {
    nyhetssakService
      .getSakBruker(this.props.match.params.id)
      .then(saker => (this.saker = saker))
      .catch((error: Error) => Alert.danger(error.message));
  }
}