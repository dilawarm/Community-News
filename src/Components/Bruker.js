import * as React from 'react';
import Button from 'react-bootstrap/Button';
import Form from 'react-bootstrap/Form';
import { Component } from 'react-simplified';
import {Alert} from '../widgets';
import {Bruker, brukerService} from '../services';
import {Navigation} from './staticComponents';
import { createHashHistory } from 'history';

const history = createHashHistory();

export class Login extends Component {
  brukernavn = '';
  passord = '';
  inn_bruker = null;

  render() {
    return <>
    <Form>
  <Form.Group controlId="exampleForm.ControlInput1">
    <Form.Label>Brukernavn</Form.Label>
    <Form.Control 
    type="text" 
    value={this.brukernavn}
    onChange={(event: SyntheticInputEvent<HTMLInputElement>) => (this.brukernavn = event.target.value)}
    />
  </Form.Group>
  <Form.Group controlId="exampleForm.ControlInput1">
    <Form.Label>Passord</Form.Label>
    <Form.Control
    type="password" 
    value={this.passord}
    onChange={(event: SyntheticInputEvent<HTMLInputElement>) => (this.passord = event.target.value)}
    />
  </Form.Group>
  <Button variant="primary" onClick={this.logg_in}>Logg inn</Button>
</Form>
    </>
  }

  logg_in() {
    brukerService
      .login(this.brukernavn, this.passord)
      .then(json => {
    	  localStorage.token = json.jwt;
    	  console.log(JSON.stringify(json));
        brukerService
          .getOne(this.brukernavn)
          .then(json => {
            console.log(json.brukernavn);
            console.log(json.brukerId);
            //ny_bruker = new Bruker(json.br)
            let navigation = Navigation.instance();
            navigation.mounted(new Bruker(json.brukerId, json.brukernavn));
        })
          .then(() => {
            brukerService
              .postToken(localStorage.token)
              .then(console.log("token posted"))
              .catch((error: Error) => Alert.danger(error.message));
          })
          .catch((error: Error) => Alert.danger(error.message));
     })
     .then(history.push("/"))
     .then(Alert.success("Nå er du logget inn!"))
     .catch((error: Error) => Alert.danger(error.message));
  } 
}

export class Registrer extends Component {
  brukernavn = '';
  passord = '';

  render() {
    return <>
    <Form>
  <Form.Group controlId="exampleForm.ControlInput1">
    <Form.Label>Brukernavn</Form.Label>
    <Form.Control 
    type="text" 
    value={this.brukernavn}
    onChange={(event: SyntheticInputEvent<HTMLInputElement>) => (this.brukernavn = event.target.value)}
    />
  </Form.Group>
  <Form.Group controlId="exampleForm.ControlInput1">
    <Form.Label>Passord</Form.Label>
    <Form.Control
    type="password" 
    value={this.passord}
    onChange={(event: SyntheticInputEvent<HTMLInputElement>) => (this.passord = event.target.value)}
    />
  </Form.Group>
  <Button variant="success" onClick={this.register}>Registrer bruker</Button>
</Form>
    </>
  }

  register() {
    brukerService
      .registrer(this.brukernavn, this.passord)
      .then(history.push("/"))
      .then(Alert.success("Brukeren er registrert! Du kan logge inn nå."))
      .catch((error: Error) => console.log("")); 
  }
}