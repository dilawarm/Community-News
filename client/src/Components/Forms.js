// @flow

import * as React from 'react';
import Button from 'react-bootstrap/Button';
import Form from 'react-bootstrap/Form';
import {Component} from 'react-simplified';
import {Alert} from '../widgets';
import {Nyhetssak, nyhetssakService, Bruker} from '../services';
import {Navigation} from './staticComponents';
import {Sak} from './Sak';
import { createHashHistory } from 'history';

const history = createHashHistory();

let kategorier = ["Nyheter", "Sport", "Kultur", "Annet"];

export class EditSak extends Component<{ match: { params: { id: number, kategori: string } } }> {
  sak: Nyhetssak | any = null;

  render() {
    if (this.sak) {
      console.log(this.sak);
    return <>
    <div className="mx-auto w-75">
    <Form>
  <Form.Group controlId="exampleForm.ControlInput1">
    <Form.Label>Overskrift</Form.Label>
    <Form.Control 
    type="text" 
    value={this.sak.overskrift}
    onChange={(event: SyntheticInputEvent<HTMLInputElement>) => (this.sak.overskrift = event.target.value)}
    />
  </Form.Group>
  <Form.Group controlId="exampleForm.ControlTextarea1">
    <Form.Label>Innhold</Form.Label>
    <Form.Control as="textarea" rows="9"
    type="text" 
    value={this.sak.innhold}
    onChange={(event: SyntheticInputEvent<HTMLInputElement>) => (this.sak.innhold = event.target.value)}
    />
  </Form.Group>
  <Form.Group controlId="exampleForm.ControlInput1">
    <Form.Label>Bilde</Form.Label>
    <Form.Control
    type="text" 
    placeholder="Lim inn URL"
    value={this.sak.bilde}
    onChange={(event: SyntheticInputEvent<HTMLInputElement>) => (this.sak.bilde = event.target.value)}
    />
  </Form.Group>
  <Form.Group controlId="exampleForm.ControlSelect2">
    <Form.Label>Kategori</Form.Label>
    <br></br>
    <select
    type="text"
    value={this.sak.kategori}
    onChange={(event: SyntheticInputEvent<HTMLInputElement>) => (this.sak.kategori = event.target.value)}>
    {kategorier.map(kategori => (
      <option>{kategori}</option>
    ))}
    </select>
  </Form.Group>
  <Form.Group controlId="exampleForm.ControlSelect2">
    <Form.Label>Viktig?</Form.Label>
    <br></br>
    <input
    type="checkbox"
    checked={this.sak.viktighet}
    onChange={(event: SyntheticInputEvent<HTMLInputElement>) => (this.sak.viktighet = event.target.checked)}
    />
  </Form.Group>
  <Button variant="success" onClick={this.save}>Endre nyhetsartikkel</Button>
</Form>
</div>
    </>
    }
    else {
      return (
        <div>Laster redigeringsskjema</div>
      )
    }
  }

  mounted() {
    nyhetssakService
      .getSakKatId(this.props.match.params.kategori, this.props.match.params.id)
      .then(sak => (this.sak = sak))
      .catch((error: Error) => Alert.danger(error.message));
  }

  save() {
    nyhetssakService
      .updateSak(this.props.match.params.id, this.sak.overskrift, this.sak.innhold, this.sak.bilde, this.sak.kategori, this.sak.viktighet)
      .then(() => {
        if (this.sak) {
          history.push("/");
          Alert.success("Saken er oppdatert!");
        }
      })
      .catch((error: Error) => Alert.danger(error.message));
  }
}

export class AddSak extends Component {
  overskrift: string = '';
  innhold: string = '';
  bilde: string = '';
  kategori: string = kategorier[0];
  viktighet: boolean = false;
  sak: Nyhetssak | any = null;
  inn_bruker: Bruker | any = null;

  render() {
    return <>
    <div className="mx-auto w-75">
    <Form>
  <Form.Group controlId="exampleForm.ControlInput1">
    <Form.Label>Overskrift</Form.Label>
    <Form.Control 
    type="text" 
    value={this.overskrift}
    onChange={(event: SyntheticInputEvent<HTMLInputElement>) => (this.overskrift = event.target.value)}
    />
  </Form.Group>
  <Form.Group controlId="exampleForm.ControlTextarea1">
    <Form.Label>Innhold</Form.Label>
    <Form.Control as="textarea" rows="9"
    type="text" 
    value={this.innhold}
    onChange={(event: SyntheticInputEvent<HTMLInputElement>) => (this.innhold = event.target.value)}
    />
  </Form.Group>
  <Form.Group controlId="exampleForm.ControlInput1">
    <Form.Label>Bilde</Form.Label>
    <Form.Control
    placeholder="Lim inn URL"
    type="text" 
    value={this.bilde}
    onChange={(event: SyntheticInputEvent<HTMLInputElement>) => (this.bilde = event.target.value)}
    />
  </Form.Group>
  <Form.Group controlId="exampleForm.ControlSelect2">
    <Form.Label>Kategori</Form.Label>
    <br></br>
    <select
    type="text"
    value={this.kategori}
    onChange={(event: SyntheticInputEvent<HTMLInputElement>) => (this.kategori = event.target.value)}>
    {kategorier.map(kategori => (
      <option>{kategori}</option>
    ))}
    </select>
  </Form.Group>
  <Form.Group controlId="exampleForm.ControlSelect2">
    <Form.Label>Viktig?</Form.Label>
    <br></br>
    <input  
    type="checkbox"
    value={this.viktighet}
    onChange={(event: SyntheticInputEvent<HTMLInputElement>) => (this.viktighet = event.target.checked)}
    cheked={this.viktighet}
    />
  </Form.Group>
  <Button variant="success" onClick={this.add}>Legg til nyhetsartikkel</Button>
</Form>
</div>
    </>
  }

  add() {
    nyhetssakService
      .postSak(this.overskrift, this.innhold, this.bilde, this.kategori, this.viktighet, this.inn_bruker.brukerId)
      .then(() => {
        history.push("/");
        Alert.success("Sak opprettet!");
      })
      .catch((error: Error) => Alert.danger(error.message));
  }

  mounted() {
    let nav: any = Navigation.instance()
    this.inn_bruker = nav.inn_bruker;
  }
}