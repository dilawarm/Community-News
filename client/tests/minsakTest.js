// @flow

import * as React from 'react';
import { Component } from 'react-simplified';
import {MineSaker} from '../src/Components/Views';
import { shallow, mount } from 'enzyme';
import { Nyhetssak, nyhetssakService } from '../src/services.js';

//jest.mock('../src/services.js');

describe('MineSaker tests', () => {
    const wrapper = shallow(<MineSaker match={{params: {id: 1}}}/>);

    it('initially', () => {
        let instance = MineSaker.instance();
        expect(typeof instance).toEqual('object');
        jest.spyOn(nyhetssakService, 'getSakBruker').mockResolvedValue([]);
        wrapper.update();
        if (instance) expect(wrapper.debug()).toMatchSnapshot();
    });

    it('after load', () => {
        let tab: Nyhetssak[] = [new Nyhetssak(1, 'Overskrift', 'Innhold', '2019-10-11 02-20', 'bilde.jpg', 'Sport', false, 10, 1)]
        jest.spyOn(nyhetssakService, 'getSakBruker').mockResolvedValue(tab);
        wrapper.update()
        let instance = MineSaker.instance();
        expect(typeof instance).toEqual('object');
        if (instance) {
            instance.forceUpdate();
            instance.saker = tab;
            expect(wrapper.debug()).toMatchSnapshot();
        };
    });
});