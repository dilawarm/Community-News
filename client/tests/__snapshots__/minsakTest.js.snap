// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MineSaker tests after load 1`] = `
"<Fragment>
  <div className=\\"mx-auto w-75\\">
    <a href=\\"#/kategori/Sport/1\\">
      <Card body={false}>
        <img src=\\"bilde.jpg\\" width=\\"945\\" height=\\"400\\" />
        <CardBody>
          <CardTitle>
            Overskrift
          </CardTitle>
          <CardText>
            <p id=\\"tid\\">
              Sist oppdatert: 
              2019-10-11 02-20
            </p>
          </CardText>
        </CardBody>
      </Card>
    </a>
  </div>
</Fragment>"
`;

exports[`MineSaker tests initially 1`] = `"<Fragment />"`;
